runtime/src/bank.rs

```diff
@@ -1639,6 +1640,14 @@ impl Bank {
             .fetch_add(collected_rent, Ordering::Relaxed);
     }
 
+    fn run_incinerator(&self) {
+        if let Some((account, _)) = self.get_account_modified_since_parent(&sysvar::incinerator::id()) {
```

Comment:
Turns out I can't use the bank `get_sysvar_account`/`update_sysvar_account` machinery at all.   The incinerator is unlike all other sysvars in this important way:
All other sysvars are updated by the bank before processing the slot, and may not be updated by any instruction within that slot (hence `get_sysvar_account` skips the current slot when looking for the current sysvar's value).  The incinerator is the opposite -- it can only be updated by an instruction within the slot, and is deallocated at freeze.